### ISSUE TITLE
Demand Sensing Data is not Visible in Looker Dashboard

### ISSUE THREAD

**Issue by Gururaja12345** [looker-open-source/block-cortex-sap#33]
HI

I have deployed Demand sensing test data from SAP and i am not able to see data in Looker Dashboard.
Please find the below Input details which i deployed :

Forecast Horizon (weeks) :13
Content Window (weeks): 52
Model Training Budget (hours): 1
Deploy Test data : Yes
Mandt Client : 900

Could someone help me why i am unable to see data in Looker Dashboard for Demand sensing.
![Demand sensing Data set](https://github.com/looker-open-source/block-cortex-sap/assets/135985089/a4a5aac9-8625-4e6c-9bd0-d5d68b65ce71)
![Demand sensing](https://github.com/looker-open-source/block-cortex-sap/assets/135985089/350829b0-be75-46bd-99ed-c65a62b1d66d)
I have deployed Demand sensing and i have data set. Please find the attachement and i cannot see any data in Looker

**Comment by benschuler** [comment]
Hi @Gururaja12345 , can you please confirm that you have maintained the required customizations in the Demand Sensing dashboards as described in the corresponding repository: https://github.com/looker-open-source/block-cortex-demand-sensing#required-customizations

If yes, can you please share a screenshot of the manifest file for us to review?

**Comment by Gururaja12345** [comment]
Hi Benschuler,

Is it possible to connect over Google meet so i can send meeting invite.

**Comment by Gururaja12345** [comment]
Hi Benschuler,

How to share Manifest file ? Could you please guide so that i can follow the same steps and share it to you .

Many Thanks

**Comment by benschuler** [comment]
On your dashboard, click the three dots at the top right corner, it should open a menu. Select the menu entry "Go to LookML". This should open the file browser. Inside the folder structure on the left, you should see the `manifest.lkml` file. Click on it to open and review the values that are maintained there. You need to specify the connection name to BigQuery as it is maintained in your Looker instance, as well as the project ID and dataset and the client where the data is stored. You can review a sample screenshot below. Your values will be different, so make sure you are maintaining the right values here that match your deployment parameters.

![image](https://github.com/looker-open-source/block-cortex-sap/assets/56956081/0910226f-9fea-43a7-9e18-fcc7f8aca5a3)

**Comment by Gururaja12345** [comment]
HI Benschuler,

I tried changing the values but i dont have option to EDIT is it ADMIN Access is required.
![Screenshot 2023-07-03 175528](https://github.com/looker-open-source/block-cortex-sap/assets/135985089/8f3e51a6-7082-48bf-ba40-91e00944d009)

**Comment by benschuler** [comment]
I can see that your dashboards are installed as a Looker Block from the marketplace, which means that the files cannot be edited directly, instead someone with the appropriate permissions needs to maintain the configuration as described in this help article: https://cloud.google.com/looker/docs/marketplace#managing_installed_tools

Please reach out to the team that has installed the blocks for you and ask them to maintain the correct values based on your BigQuery deployment, afterwards the dashboards should show the data.

**Comment by Gururaja12345** [comment]
Hi Benschuler,

we can't modify anything on the current Looker Blocks, instead of deploying directly from GitHub (the current way it is configured) we should make a fork of the GitHub repo and build off of that. This allows us to make our own modifications and not be locked in.

Could you please help me how to Deploy Demand sensing Data (step by step process) 

Many Thanks in Advance.

**Comment by benschuler** [comment]
Hi @Gururaja12345 ,

If you want to deploy the dashboards directly from a repository instead of doing a marketplace installation, here is how I would recommend you to do it:

1) Create your own fork of the Github repository that you want to install. I assume you want this one --> https://github.com/looker-open-source/block-cortex-demand-sensing

2) Create a blank project in Looker --> https://cloud.google.com/looker/docs/create-projects#creating_a_blank_project

3) In the new empty project, go to Git settings and maintain the repository fork that you have created in step 1) --> https://cloud.google.com/looker/docs/setting-up-git-connection

4) Now you should be able to make any necessary changes directly in the Looker instance and send those changes back to your own repository via the built-in git tools --> https://cloud.google.com/looker/docs/version-control-and-deploying-changes

5) Make sure you have made all required changes that are described in the readme file of the dashboards --> https://github.com/looker-open-source/block-cortex-demand-sensing#required-customizations

Please feel free to let us know if you have any further issues.

**Comment by Gururaja12345** [comment]
Hi Benschuler,

We have followed the above mention steps and we have warnings when we Validate the LookML. Please find the below screenshot and let me know.

![Looker Validation](https://github.com/looker-open-source/block-cortex-sap/assets/135985089/0cf0873f-f2f7-4455-b922-cd9773daa14d)

**Comment by benschuler** [comment]
From your screenshot it is a bit hard to identify the full scope of your warnings, but from what I can see on the screen your Looker instance is complaining about values that are actually valid: https://cloud.google.com/looker/docs/reference/param-lookml-dashboard#field

I checked in our internal instance where these fields exist and we are not getting any warnings, so I suggest to get a Looker expert to help you troubleshoot this behavior. If they find that there are some issues with our dashboards, we can work with them/you to get them resolved, but they need to identify the root cause first before we can suggest any further steps here.

**Comment by Gururaja12345** [comment]
Hi Ben,

The dashboards have been configured properly but the underlying issue appears to be related to the data being out of date range of the dashboards. We heard that you have modified your demo data to stay 'current' via SQL queries. 
Is it possible for you to share this with us, so that we can do the same?

**Comment by Gururaja12345** [comment]
Hi Ben,
We have corrected the Lookml dashboard issues but still we cannot see data in Dashboards for Demand Sensing Data.
The dashboards have been configured properly but the underlying issue appears to be related to the data being out of date range of the dashboards. 
We heard that you have modified your demo data to have the dates stay 'current' via SQL queries. Is it possible for you to share this with us, so that we can do the same?
Thank you Kindly.

**Comment by benschuler** [comment]
If you want to use the static sample data set and get the data to move forward into the future for demo purposes, one option is to create a dedicated dataset in BigQuery where you can then create new views that point to the original tables in your reporting dataset, but with the following adjustments:

For date fields, you can take advantage of the `DATE_DIFF` function in BigQuery ([details here](https://cloud.google.com/bigquery/docs/reference/standard-sql/date_functions#date_diff)) and use that in combination with the `CURRENT_DATE` and `DATE_ADD` functions to create a statement like this one:

```
SELECT 
    * EXCEPT (
        WeekStart
    )
    , DATE_ADD(WeekStart, INTERVAL DATE_DIFF(CURRENT_DATE(), "2022-01-01", WEEK) WEEK) AS WeekStart
FROM
    `SAP_REPORTING.DemandPlan`
```

Create a similar view for all tables/views that are referenced in the Demand Sensing dashboards and make sure you are maintaining the same start date (in my example `2022-01-01`) for all functions, otherwise your data will be all over the place. For some fields, you will have to use DAY interval instead of WEEK, so make sure you are properly testing all of the adjustments and carefully read the instructions in the suggested help article to be aware of any prerequisites for these date functions.

Finally, in your Looker manifest, maintain the new dataset as the source dataset for your dashboards, and then it should show the data shifted by the number of days/weeks that you specified.

Please bear in mind that this is only a suggestion to make static demo data look more alive, it is not something that any real implementation would require, because in a real implementation new data would be flowing in on a daily/weekly basis.

**Comment by benschuler** [comment]
Hi ,

I hope that the details provided in my previous response were helpful. Since we have not heard anything back for a while, I assume you were able to get the issues resolved and we can close this issue. If you have any further questions, please feel free to open a new issue and we will be happy to assist you.
